openquake/hazardlib/gsim/can15/eastern.py

```diff
@@ -0,0 +1,195 @@
+"""
+:module:`openquake.hazardlib.gsim.nrcan15` implements
+:class:`EasternCan15Mid`, :class:`EasterCnan15Low`,
+:class:`EasternCan15Upp`
+"""
+
+import re
+import numpy as np
+
+from openquake.hazardlib.gsim.base import CoeffsTable
+from openquake.hazardlib.gsim.can15 import utils
+from openquake.hazardlib.gsim.can15.western import get_sigma
+
+from openquake.hazardlib.imt import PGA
+from openquake.hazardlib.const import StdDev
+from openquake.hazardlib.gsim.pezeshk_2011 import PezeshkEtAl2011
+from openquake.hazardlib.gsim.boore_atkinson_2011 import Atkinson2008prime
+from openquake.hazardlib.gsim.atkinson_boore_2006 import \
+    AtkinsonBoore2006Modified2011
+from openquake.hazardlib.gsim.silva_2002 import (
+    SilvaEtAl2002DoubleCornerSaturation, SilvaEtAl2002SingleCornerSaturation)
+
+
+class EasternCan15Mid(PezeshkEtAl2011):
+    """
+    Implements the hybrid GMPE used to compute hazard in the Eastern part of
+    Canada.
+
+    The GMPEs used are:
+
+    - Pezeshk et al. (2011) - For this GMPE we scale the ground motion from
+    hard rock to B/C using the correction proposed in Atkinson and Adams (2013)
+    Table 2 page 994. As the distance metric used is Rrup we compute and
+    equivalent Rrup distance from Repi using the equations in Appendix A (page
+    31) of Atkinson (2012).
+
+    - Atkinson (2008) as revised in Atkinson and Boore (2011). As the distance
+    metric used is Rjb we compute and equivalent Rjb distance from Repi
+    using the equations in Appendix A (page 31) of Atkinson (2012).
+
+    - Atkinson and Boore (2006) as revised in Atkinson and Boore (2011). As
+    the distance metric used is Rjb we compute and equivalent Rjb distance
+    from Repi using the equations in Appendix A (page 31) of Atkinson (2012).
+
+    - Silva et al. (2002) single corner and saturation.
+
+    - Silva et al. (2002) double corner and saturation.
+    """
+
+    # Required site parameters
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    # Required distance is only repi since rrup and rjb are obtained from repi
+    REQUIRES_DISTANCES = set(('repi',))
+
+    # Set rake
+    REQUIRES_RUPTURE_PARAMETERS = set(('rake', 'mag'))
+
+    # Distances to be excluded while checking this GMPE
+    DO_NOT_CHECK_DISTANCES = set(('rrup', 'rjb'))
+
+    def apply_correction_to_BC(self, mean, imt, dists):
+        """
+        """
+        if re.search('^SA', str(imt)):
+            cff = self.SITE_COEFFS[imt]
```

Comment:
You can just use

```python
if imt.period:
  ...
else:
 ...
```
an AttributeError will be raised if the IMT is not SA or PGA which should be good enough.